### ISSUE TITLE
count() value is correct, but get() returns less or no items in the collection

### ISSUE THREAD

**Issue by rudolphp** [babenkoivan/scout-elasticsearch-driver#256]
The $model->search('dummy')....->count() returns a count which matches with what is returned as hits in Kibana. But when I replace 'count' by 'get' the Collection items is empty.
(more info - searchRules and Filter is being used too)

**Comment by iget-esoares** [comment]
Please provide more information so we can try to reproduce it here.

The `...` part of your code IS relevant.

**Comment by rudolphp** [comment]
Filter is
`$r = $model::search('dummy')->whereIn('district', request('d'));
`
SearchRule is
`
return [
            'should' => [
                [
                    'match_phrase_prefix' => [
                        'surname' => [
                            'query' =>  request('ln') ? request('ln') : '',
                        ],
                    ],
                ],
                [
                    'match_phrase_prefix' => [
                        'forename1' => [
                            'query' => request('fn') ? request('fn') : '',
                        ],
                    ],
                ],      
                [
                    'match_phrase_prefix' => [
                        'forename2' => [
                            'query' => request('fn') ? request('fn') : '',
                        ],
                    ],
                ],
                [
                    'match_phrase_prefix' => [
                        'forename3' => [
                            'query' => request('fn') ? request('fn') : '',
                        ],
                    ],
                ],
                [
                    'match_phrase_prefix' => [
                        'mothersmaiden' => [
                            'query' => request('mn') ? request('mn') : '',
                        ],
                    ],
                ]
            ]
        ];
`

Mapping in model is
`
protected $mapping = [
        "include_type_name" => true,
        "properties"=> [
            "surname"=>    [ "type"=> "text", "fielddata"=> true ], 
            "forename1"=>     [ "type"=> "text", "fielddata"=> true  ], 
            "forename2"=>      [ "type"=> "text", "fielddata"=> true ],
            "forename3"=>      [ "type"=> "text", "fielddata"=> true ],
            "mothersmaiden"=>      [ "type"=> "text", "fielddata"=> true ],
            "mothersmaiden2"=>      [ "type"=> "text", "fielddata"=> true ],
            "district"=>      [ "type"=> "text", "fielddata"=> true ],
            "page"=>      [ "type"=> "integer" ],
            "event"=>      [ "type"=> "text" ],
            "quarter"=>      [ "type"=> "integer" ],
            "year"=>      [ "type"=> "integer" ],
            "pagenumber"=>      [ "type"=> "integer" ],
            "letter"=>      [ "type"=> "text" ],
            "imgref"=>      [ "type"=> "text" ],
            "lsurname"=>      [ "type"=> "text" ],
            "lforename1"=>      [ "type"=> "text" ],
            "lforename2"=>      [ "type"=> "text" ],
            "lforename3"=>      [ "type"=> "text" ],
            "ldistrictvolumeno"=>      [ "type"=> "text" ],
            "lpage"=>      [ "type"=> "integer" ],
            "linktoimages"=>      [ "type"=> "text" ]
        ]
    ];
`

**Comment by alvarezmario** [comment]
I'm having the same issue:
```
Institution::search($request->term)->get()
```
returns an empty array, but counts works, and I tried the `raw` method and the query is returning data, it seems that what is failing is when building the Eloquent models..

**Comment by babenkoivan** [comment]
@alvarezmario @rudolphp what version gives you incorrect result?

**Comment by alvarezmario** [comment]
@babenkoivan version 4.0

**Comment by rudolphp** [comment]
@babenkoivan using the latest version ^4.0

**Comment by babenkoivan** [comment]
> it seems that what is failing is when building the Eloquent models..
@alvarezmario @rudolphp I've got bunch of questions:
1. Are you using custom Scout key for models (`Model::getScoutKeyName` or `Model::getScoutKey`)? 
2. Are you using `Builder::select` method? (`Model::search('foo')->select(['field1', 'field2'])`)?
3. Are soft deletes enabled in the Scout config? 
4. What type of `indexer` do you use (single or bulk)?
5. Are you using `Model::toSearchbleArray`? If so, what is the content?

**Comment by rudolphp** [comment]
> 1. Are you using custom Scout key for models (`Model::getScoutKeyName` or `Model::getScoutKey`)?

No

> 2. Are you using `Builder::select` method? (`Model::search('foo')->select(['field1', 'field2'])`)?

No

> 3. Are soft deletes enabled in the Scout config?

No

> 4. What type of `indexer` do you use (single or bulk)?

I indexed using logstash. I am not using this config, so it must be default 'single'

> 5. Are you using `Model::toSearchbleArray`? If so, what is the content?

No

**Comment by zhouzishu** [comment]
I meet the same problem.
I try to find the where the problem is, I find that `$results['hits']['total']['value']` is unset.

**Comment by erashdan** [comment]
I had same issue but fixed after I upgrade elasticsearch to version 7.1

**Comment by zhouzishu** [comment]
> I had same issue but fixed after I upgrade elasticsearch to version 7.1

Yeah, I have fixed by this way. I think it is a compatibility problem. The package author has noted in readme.md

**Comment by itguy614** [comment]
I am running ES 7.3 and running into this problem.  If I use a ->raw() instead of ->get() I properly get my results.

**Comment by chrisgrim** [comment]
I am having this issue too. On ES 7.3 as well. Is there a way to fix this?

**Comment by chrisgrim** [comment]
Actually I figured out how to fix this for me. I had to run php artisan scout:import "App\MyEvent" and it worked for me

**Comment by babenkoivan** [comment]
> I try to find the where the problem is, I find that $results['hits']['total']['value'] is unset.

Interesting finding. I tried to reproduce the issue again using different versions of ES, but no luck. It looks like the issue is related to some specific version of ES or configuration as not many people complain about it.

**Comment by vpillinger** [comment]
The package gets the models from ElasticSearch, those models are then loaded by Laravel to serialize them in output. For example, if elastic search returns model ID 1, 2, and 3 from the database and you return that in a JSON array. Then models 1, 2, and 3 are loaded from the DB and serialized. If one or more of those models cannot be loaded from the DB (due to an error, incorrect id, permission issue, ect.) then they are excluded from the output.

If you have out of date data or corrupted data in ES (say after an update) then its possible to have nothing load with a silent error.

**Comment by rognales** [comment]
Same issue with ES 7.1.0 & 7.4.0 and package ^4.0.

->raw() returns expected result but ->get() returns empty.

**Comment by rw4lll** [comment]
Faced with the same problem. I downgraded to **v3.12.0** and it's solve the problem.

**Comment by chistel** [comment]
> 
> 
> > I try to find the where the problem is, I find that $results['hits']['total']['value'] is unset.
> 
> Interesting finding. I tried to reproduce the issue again using different versions of ES, but no luck. It looks like the issue is related to some specific version of ES or configuration as not many people complain about it.

am faced with the same issue, after a fresh installation of this package

**Comment by chistel** [comment]
> 
> 
> > > I try to find the where the problem is, I find that $results['hits']['total']['value'] is unset.
> > 
> > 
> > Interesting finding. I tried to reproduce the issue again using different versions of ES, but no luck. It looks like the issue is related to some specific version of ES or configuration as not many people complain about it.
> 
> am faced with the same issue, after a fresh installation of this package

never mind, I was using elasticsearch version 6

**Comment by tabjen99** [comment]
I have a similar issue , It's due to records in your database has been deleted.But somehow it doesn't update this delete action to ES. when u perform search , ES return records that doesn't exsit in database, hence it's showing less items in collections.

**Comment by OscarKolsrud** [comment]
> I have a similar issue , It's due to records in your database has been deleted.But somehow it doesn't update this delete action to ES. when u perform search , ES return records that doesn't exsit in database, hence it's showing less items in collections.

This 100%

**Comment by jalil-ashraf** [comment]
> The $model->search('dummy')....->count() returns a count which matches with what is returned as hits in Kibana. But when I replace 'count' by 'get' the Collection items is empty.
> (more info - searchRules and Filter is being used too)

Can you please tell me how u solved this issue?

**Comment by theojlab** [comment]
Inspired by tabjen99's comment, I deleted the ElasticSearch index and then performed a fresh scout:import which resolved the problem.
